src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -4648,6 +4648,12 @@ and TcAnonRecdType cenv newOk checkConstraints occ env tpenv isStruct args m =
     let unsortedFieldIds = args |> List.map fst |> List.toArray
     let anonInfo = AnonRecdTypeInfo.Create(cenv.thisCcu, tupInfo, unsortedFieldIds)
 
+    // Check for duplicate field IDs
+    unsortedFieldIds
+    |> Array.countBy (fun fieldId -> fieldId.idText)
+    |> Array.iter (fun (idText, count) ->
```

Comment:
Minor nit - we generally minimise use of `Array.iter` and `List.iter` in the codebase in favour of `for` loops - which are generally clearer.  

So maybe
```fsharp
    // Check for duplicate field IDs
    for (idText, count) in unsortedFieldIds |> Array.countBy (fun id -> id.idText) do
        if count > 1 then
            errorR (Error (FSComp.SR.tcAnonRecdDuplicateFieldId(idText), m))
```

And may as well multi-line format the `if` since future auto-source-code-formatting is certainly going to do it anyway.

Reply:
Hmm... I can change it to a for loop, it will also be a bit clunky IMO.

Another option:
```fsharp
unsortedFieldIds
|> Array.countBy (fun fieldId -> fieldId.idText)
|> Array.tryFind (fun (_, count) -> count > 1)
|> function
    | None -> ()
    | Some (idText, _) -> error (Error (FSComp.SR.tcAnonRecdTypeDuplicateFieldId(idText), m))
```
but not sure if that's better than the original. Last function could be replaced with `Option.iter` for conciseness

Reply:
`Array.iter` is fine in this case.